src/components/navbar/SearchBar.tsx

```diff
@@ -7,48 +7,74 @@ import {
   InputLabel,
   OutlinedInput,
   useMediaQuery,
-  useTheme,
+  useTheme
 } from "@mui/material";
 import { makeStyles } from "@mui/styles";
-import React from "react";
+import React, { useCallback, useEffect, useState } from "react";
+import theme from "styles/theme";
+import gaSearchKey from "utils/gaSearchKey";
 
 // Class names from https://stackoverflow.com/questions/58963242/change-border-color-on-material-ui-textfield
-const useStyles = makeStyles((theme) => ({
+const useStyles = makeStyles({
   form: {
     margin: "16px 0px 16px 16px",
     minWidth: "30%",
     "& .MuiInputLabel-outlined.Mui-focused, .MuiInputLabel-outlined": {
       marginLeft: -8,
       [theme.breakpoints.down("mobileCard")]: {
-        marginLeft: 0,
-      },
+        marginLeft: 0
+      }
     },
     "& .MuiInputLabel-outlined.Mui-focused": {
-      color: "#DCDCFF",
+      color: "#DCDCFF"
     },
     "& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline": {
-      borderColor: "#DCDCFF",
+      borderColor: "#DCDCFF"
     },
     "&:hover .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline": {
-      borderColor: "rgba(255, 255, 255, 0.50)",
-    },
-  },
-}));
+      borderColor: "rgba(255, 255, 255, 0.50)"
+    }
+  }
+});
+
+interface Props {
+  handleSearch: (term: string) => void;
+}
 
-const SearchBar = ({ handleChange, onChange }) => {
+const SearchBar = ({ handleSearch }: Props) => {
   const theme = useTheme();
   const mobile = useMediaQuery(theme.breakpoints.down("mobileCard"));
-
   const classes = useStyles();
 
+  const [searchInput, setSearchInput] = useState("");
+
+  useEffect(() => {
+    searchInput ? handleSearch(searchInput) : handleSearch("");
+  }, []);
```

Comment:
Just something I noticed, I don't believe this is currently used as there's no way to set `searchInput` on page load. I bet this could be done with react-router and getting a query from the url such as `https://shouldiplaythis.com/?query=metroid'

Reply:
I implemented this while testing things in development. I noticed that when I would re-render the page while making dev changes it wouldn't use what I already had in my search bar. You're likely correct that it doesn't actually get used in the real world.

I do however plan on implementing query params in the future, so I'll leave it be for now!